### PR TITLE
dontTouch: Skip probe and properties.

### PR DESCRIPTION
**Pull request by dtzSiFive**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?
  - (None, title seems sufficient?)

#### Type of Improvement

- API modification
- Bugfix

#### Desired Merge Strategy

- Squash: The PR will be squashed and merged (choose this if you have no preference).

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels? (Select the most appropriate one based on the "Type of Improvement")
- [ ] Did you mark the proper milestone (Bug fix: `3.5.x`, `3.6.x`, or `5.x` depending on impact, API modification or big change: `6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by jackkoenig** [comment]
I see you marked this API modification and BugFix, I went with BugFix for the label (which is used by release notes) because it seems better for `dontTouch` of something with Probes or Properties in it to "work" (for some definition of work) rather than crash. The API modification is something that used to crash now does something.

Happy to change it if @dtzSiFive or others disagree though.

**Comment by dtzSiFive** [comment]
Okay, I agree with all that.  I can drop the "API modification" part, but no objection to your labeling or reasoning. :+1: .

File: core/src/main/scala/chisel3/dontTouch.scala [hunk 1]
```diff
@@ -34,6 +36,8 @@ object dontTouch {
   def apply[T <: Data](data: T): T = {
     requireIsHardware(data, "Data marked dontTouch")
     data match {
+      case d if DataMirror.hasProbeTypeModifier(d) => ()
+      case _:   Property[_] => ()
```
Comment (seldridge) [inline core/src/main/scala/chisel3/dontTouch.scala:40]: Explicitly returning `Unit`, i.e. `()`, is probably best here. Most places in the code base will not make this explicit and just put no text after the `=>`. I like the style here of `()`, though.

**Review by seldridge — APPROVED** [review]

**Review by mikeurbach — COMMENTED** [review]
This makes sense to me, thanks for include Property along with Probe. If the others are on board, I am on board with this.

File: core/src/main/scala/chisel3/dontTouch.scala (hunk 1 above)
Comment (mikeurbach) [reply core/src/main/scala/chisel3/dontTouch.scala:40]: 👍 i agree that i like explicitly returning Unit

Comment (dtzSiFive) [reply core/src/main/scala/chisel3/dontTouch.scala:40]: (Oh, neat! TIL, thanks!)

**Comment by dtzSiFive** [comment]
Thanks, all!

**Pull request merged**